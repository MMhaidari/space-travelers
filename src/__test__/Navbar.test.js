import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/Navbar';
import '@testing-library/jest-dom';

jest.mock('react-bootstrap/esm/Navbar', () => 'MockedBootstrapNavbar');
describe('NavBar', () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
  });

  it('renders a logo', () => {
    const logo = screen.getByAltText('header logo');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('src', 'logo.png');
  });

  it('renders a link to the Rockets page', () => {
    const rocketsLink = screen.getByText('Rockets');
    expect(rocketsLink).toBeInTheDocument();
    expect(rocketsLink).toHaveAttribute('href', '/');
  });

  it('renders a link to the Missions page', () => {
    const missionsLink = screen.getByText('Mission');
    expect(missionsLink).toBeInTheDocument();
    expect(missionsLink).toHaveAttribute('href', '/missionpage');
  });

  it('renders a link to the My Profile page', () => {
    const profileLink = screen.getByText('MyProfile');
    expect(profileLink).toBeInTheDocument();
    expect(profileLink).toHaveAttribute('href', '/profilepage');
  });
  it('Navbar snapshots match', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
