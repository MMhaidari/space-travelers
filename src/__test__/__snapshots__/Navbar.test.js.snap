// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Navbar snapshots match 1`] = `
<div
  className="navbar"
>
  <div
    className="header-logo"
  >
    <img
      alt="header logo"
      className="logo"
      src="logo.png"
    />
    <h3
      className="nav-header"
    >
      space travelers' Hub
    </h3>
  </div>
  <ul
    className="nav-links"
  >
    <li
      className="nav-li"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="nav-li"
    >
      <a
        className="nav-link"
        href="/missionpage"
        onClick={[Function]}
      >
        Mission
      </a>
    </li>
    <div
      className="line"
    />
    <li
      className="nav-li"
    >
      <a
        className="nav-link"
        href="/profilepage"
        onClick={[Function]}
      >
        MyProfile
      </a>
    </li>
  </ul>
</div>
`;
